File: framework/languages/java/samples/java/TicTacToe.SessionGateway/src/main/java/systems/zlink/samples/tictactoe/sessiongateway/client/SessionActorNotificationInbox.java
package systems.zlink.samples.tictactoe.sessiongateway.client;

import java.util.ArrayList;
import java.util.List;

public final class SessionActorNotificationInbox {
    private final List<String> events = new ArrayList<>();

    public void addAll(List<String> values) {
        events.addAll(values);
    }

    public List<String> events() {
        return List.copyOf(events);
    }
}

File: framework/languages/java/samples/kotlin/TicTacToe.SessionGateway/src/main/kotlin/systems/zlink/samples/kotlin/tictactoe/sessiongateway/client/SessionActorNotificationInbox.kt
package systems.zlink.samples.kotlin.tictactoe.sessiongateway.client

class SessionActorNotificationInbox {
    private val events = mutableListOf<String>()

    fun addAll(values: List<String>) {
        events += values
    }

    fun events(): List<String> = events.toList()
}
